src/environments/examples/BitFlippingEnv.jl

```diff
@@ -0,0 +1,75 @@
+export BitFlippingEnv, GoalState
+
+"""
+In Bit Flipping Environment we have n bits. The actions are 1 to n where executing
+i-th action flips the i-th bit of the state. For every episode we sample uniformly and inital
+state as well as the target state.
+
+Refer [Hindsight Experience Replay paper](https://arxiv.org/pdf/1707.01495.pdf) for the motivation behind the environment.
+"""
+
+struct GoalState{T} <: RLBase.AbstractStateStyle end
+GoalState() = GoalState{Any}()
+
+mutable struct BitFlippingEnv <: AbstractEnv
+    N::Int
+    rng::AbstractRNG
+    state::BitArray{1}
+    goal_state::BitArray{1}
+end
+
+function BitFlippingEnv(; N = 8, rng = Random.GLOBAL_RNG)
+    state = bitrand(rng,N)
+    goal_state  = bitrand(rng,N)
+    BitFlippingEnv(N,rng,state,goal_state)
+end
+
+function BitFlippingEnv(; N = 8)
+    rng = Random.GLOBAL_RNG
+    state = bitrand(rng,N)
+    goal_state  = bitrand(rng,N)
+    BitFlippingEnv(N,rng,state,goal_state)
+end
+
+Random.seed!(env::BitFlippingEnv, s) = Random.seed!(env.rng, s)
+
+RLBase.action_space(env::BitFlippingEnv) = Base.OneTo(env.N)
+
+RLBase.legal_action_space(env::BitFlippingEnv) = Base.OneTo(env.N)
+
+function (env::BitFlippingEnv)(action::Int)
+    if 1 <= action <= env.N
+        env.state[action] = !env.state[action]
+        nothing
+    else
+        @error "Invalid Action"
+    end
+end
+
+RLBase.state(env::BitFlippingEnv) = env.state
+RLBase.state(env::BitFlippingEnv, ::GoalState) = env.goal_state
+RLBase.state_space(env::BitFlippingEnv) = Space(fill(false..true,env.N))
+RLBase.state_space(env::BitFlippingEnv, ::GoalState) = Space(fill(false..true,env.N))
+RLBase.is_terminated(env::BitFlippingEnv) = env.state == env.goal_state
+
+function RLBase.reset!(env::BitFlippingEnv)
+    env.state = bitrand(env.rng,env.N)
+    env.goal_state = bitrand(env.rng,env.N)
+end
+
+function RLBase.reward(env::BitFlippingEnv)
+    if env.state == env.goal_state
+        1.0
+    else
+        0.0
+    end
+end
+
+RLBase.NumAgentStyle(::BitFlippingEnv) = SINGLE_AGENT
+RLBase.DynamicStyle(::BitFlippingEnv) = SEQUENTIAL
+RLBase.ActionStyle(::BitFlippingEnv) = MINIMAL_ACTION_SET
+RLBase.InformationStyle(::BitFlippingEnv) = PERFECT_INFORMATION
+RLBase.StateStyle(::BitFlippingEnv) = Observation{BitArray{1}}()
```

Comment:
Since you support two state styles in this environment. You can return it here.

```suggestion
RLBase.StateStyle(::BitFlippingEnv) = (Observation{BitArray{1}}(), GoalState())
```